pontoon/base/utils.py

```diff
@@ -606,3 +609,33 @@ def is_email(email):
         return True
     except ValidationError:
         return False
+
+
+def get_project_or_redirect(
+    slug, redirect_view_name, slug_arg_name, request_user, **kwargs
+):
+    """
+    Attempts to get a project with the given slug. If the project doesn't exist, it checks if the slug is in the
+    ProjectSlugHistory and if so, it redirects to the current project slug URL. If the old slug is not found in the
+    history, it raises an Http404 error.
+    """
+    Project = apps.get_model("base", "Project")
+    ProjectSlugHistory = apps.get_model("base", "ProjectSlugHistory")
```

Comment:
Is there a reason to use this pattern rather than the function-local import in e.g. `get_download_content()` above?